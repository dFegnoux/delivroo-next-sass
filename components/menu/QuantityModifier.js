import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class QuantityModifier extends PureComponent {
  static propTypes = {
    item: PropTypes.object.isRequired,
    updateCart: PropTypes.func.isRequired,
    disabled: PropTypes.bool
  };

  static defaultProps = {
    disabled: false
  };

  render() {
    const {
      item,
      item: { quantity },
      updateCart,
      disabled
    } = this.props;

    return (
      <div className="quantityModifier">
        {!disabled && (
          <button
            className="remove"
            onClick={() => {
              updateCart(item, "remove");
            }}
          >
            -
          </button>
        )}
        <span className="quantity">{quantity}</span>
        {!disabled && (
          <button
            className="add"
            onClick={() => {
              updateCart(item, "add");
            }}
          >
            +
          </button>
        )}
      </div>
    );
  }
}

export default QuantityModifier;
