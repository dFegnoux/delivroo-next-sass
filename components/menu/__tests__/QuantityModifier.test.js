import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import QuantityModifier from "../QuantityModifier";

const props = {
  item: {
    quantity: 3
  },
  updateCart: jest.fn()
};
const container = shallow(<QuantityModifier {...props} />);

describe("QuantityModifiers snapshots", () => {
  it("should render correctly with default props", () => {
    const tree = renderer.create(<QuantityModifier {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with disabled buttons", () => {
    const tree = renderer
      .create(<QuantityModifier disabled {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Modifiers actions", () => {
  it("should correctly fired updateCart function", () => {
    container.find(".remove").simulate("click");
    expect(props.updateCart).toHaveBeenCalledWith(props.item, "remove");

    container.find(".add").simulate("click");
    expect(props.updateCart).toHaveBeenCalledWith(props.item, "add");
  });
});
