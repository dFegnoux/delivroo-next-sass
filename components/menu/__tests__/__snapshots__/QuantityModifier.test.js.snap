// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityModifiers snapshots should render correctly with default props 1`] = `
<div
  className="quantityModifier"
>
  <button
    className="remove"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="quantity"
  >
    3
  </span>
  <button
    className="add"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`QuantityModifiers snapshots should render correctly with disabled buttons 1`] = `
<div
  className="quantityModifier"
>
  <span
    className="quantity"
  >
    3
  </span>
</div>
`;
